src/components/button/button.tsx

```diff
@@ -42,38 +42,51 @@ interface CommonButtonProps {
   className?: string
 }
 
-interface ButtonAsButtonElementProps extends CommonButtonProps, ButtonHTMLAttributes<HTMLButtonElement> {
-  href?: never
-  target?: never
-  disabled?: boolean
-  onClick?: MouseEventHandler<HTMLButtonElement>
+// Define a specialized type for tertiary variant for prop removePadding
+type TertiaryButtonProps = {
+  variant: 'tertiary'
```

Comment:
`removePadding` is only available for variant `tertiary`

Reply:
**nit-pick:** per the [(draft) boolean prop interface guidelines](https://consolegroup.atlassian.net/wiki/spaces/DP/pages/3127509024/Prop+Interface+Guidelines#On-Booleans) you have access to, I think `hasNoPadding` would be preferred; `removePadding` reads like an action, not a state.